File: app/src/main/java/md/fusionworks/lifehack/util/rx/ObserverAdapter.java
package md.fusionworks.lifehack.util.rx;

import rx.Observer;

/**
 * Observer adapter that allow you to omit not needed method implementations.
 */
public abstract class ObserverAdapter<T> implements Observer<T> {
  @Override public void onCompleted() {

  }

  @Override public void onError(Throwable e) {

  }

  @Override public void onNext(T t) {

  }
}

File: app/src/main/java/md/fusionworks/lifehack/rx/ObserverAdapter.kt
package md.fusionworks.lifehack.rx

import rx.Observer

/**
 * Observer adapter that allow you to omit not needed method implementations.
 */
abstract class ObserverAdapter<T> : Observer<T> {
  override fun onCompleted() {

  }

  override fun onError(e: Throwable) {

  }

  override fun onNext(t: T) {

  }
}